server/src/test/java/org/elasticsearch/action/ingest/SimulateDocumentBaseResultTests.java

```diff
@@ -0,0 +1,143 @@
+            IllegalArgumentException e = (IllegalArgumentException) otherSimulateDocumentBaseResult.getFailure();
+            assertThat(e.getMessage(), equalTo("test"));
+        } else {
+            assertIngestDocument(otherSimulateDocumentBaseResult.getIngestDocument(), simulateDocumentBaseResult.getIngestDocument());
+        }
+    }
+
+    static SimulateDocumentBaseResult createTestInstance(boolean isFailure) {
+        SimulateDocumentBaseResult simulateDocumentBaseResult;
+        if (isFailure) {
+            simulateDocumentBaseResult = new SimulateDocumentBaseResult(new IllegalArgumentException("test"));
+        } else {
+            IngestDocument ingestDocument = createRandomIngestDoc();
+            simulateDocumentBaseResult = new SimulateDocumentBaseResult(ingestDocument);
+        }
+        return simulateDocumentBaseResult;
+    }
+
+    private static SimulateDocumentBaseResult createTestInstanceWithFailures() {
+        return createTestInstance(randomBoolean());
+    }
+
+    @Override
+    protected SimulateDocumentBaseResult createTestInstance() {
+        return createTestInstance(false);
+    }
+
+    @Override
+    protected SimulateDocumentBaseResult doParseInstance(XContentParser parser) {
+        return SimulateDocumentBaseResult.fromXContent(parser);
+    }
+
+    @Override
+    protected boolean supportsUnknownFields() {
+        return true;
+    }
+
+    @Override
+    protected Predicate<String> getRandomFieldsExcludeFilter() {
+        // We cannot have random fields in the _source field and _ingest field
+        return field ->
+            field.contains(
+                new StringJoiner(".")
+                    .add(WriteableIngestDocument.DOC_FIELD)
+                    .add(WriteableIngestDocument.SOURCE_FIELD).toString()
+            ) ||
+                field.contains(
+                    new StringJoiner(".")
+                        .add(WriteableIngestDocument.DOC_FIELD)
+                        .add(WriteableIngestDocument.INGEST_FIELD).toString()
+                );
+    }
+
+    public static void assertEqualDocs(SimulateDocumentBaseResult response, SimulateDocumentBaseResult parsedResponse) {
+        assertEquals(response.getIngestDocument(), parsedResponse.getIngestDocument());
+        if (response.getFailure() != null) {
+            assertNotNull(parsedResponse.getFailure());
+            assertThat(
+                parsedResponse.getFailure().getMessage(),
+                containsString(response.getFailure().getMessage())
+            );
+        } else {
+            assertNull(parsedResponse.getFailure());
+        }
+    }
+
+    @Override
+    public void assertEqualInstances(SimulateDocumentBaseResult response, SimulateDocumentBaseResult parsedResponse) {
+        assertEqualDocs(response, parsedResponse);
+    }
+
+    @Override
+    protected boolean assertToXContentEquivalence() {
+        return true;
+    }
```

Comment:
Maybe we can remove this and supportsUnknownFields given that we are now using the default values?